### PR TITLE
planner: refactor memory table explain information and add explain interface for other extractors

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs <crazycs520@gmail.com>

### What problem does this PR solve? 

As title said.

This PR is the first PR for the UCP issues 15378
 
### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

Release note

 - Add more explain information for memory table executor.

### CODE REVIEW

**Comment by crazycs520** [comment]
@winoros PTAL

File: planner/core/memtable_predicate_extractor.go [hunk 1]
```diff
@@ -528,6 +532,10 @@ func (e *ClusterTableExtractor) Extract(_ sessionctx.Context,
 	return remained
 }
 
+func (e *ClusterTableExtractor) explainInfo(p *PhysicalMemTable) string {
```
Comment (lonng) [inline planner/core/memtable_predicate_extractor.go]: Prefer to add a `dummyExplainInfo` instead of adding empty implementation for all extractor.

Comment (crazycs520) [reply planner/core/memtable_predicate_extractor.go]: No,becase UCP https://github.com/pingcap/tidb/issues/15378 need to implement those methods.

File: planner/core/explain.go [hunk 2]
```diff
@@ -749,41 +744,12 @@ const MetricTableTimeFormat = "2006-01-02 15:04:05.999"
...
-	}
-	if len(quantiles) == 0 {
-		quantiles = []float64{def.Quantile}
-	}
-	var buf bytes.Buffer
-	for i, quantile := range quantiles {
-		promQL := def.GenPromQL(sctx, labels, quantile)
-		if i > 0 {
-			buf.WriteByte(',')
+	explain := "table:" + p.Table.Name.O
```
Comment (winoros) [inline planner/core/explain.go]: Please use `strings.Builder`. It's more effcient to concat the string.

Comment (crazycs520) [reply planner/core/explain.go]: great, done.

**Pull request merged**